### PR TITLE
Adds save test for saveActiveFederationCreationBlockHeight

### PR DESCRIPTION
**Pull request by jeremy-then**
Add save for save active federation creation block height

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -1138,6 +1138,30 @@ void save_savePendingFederation_preRSKIP123_shouldBeSavedInStorageSerializedFrom
 
     }
 
+    @Test
+    void saveActiveFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNullAndRSKIP186IsActive() {
+        // Arrange
+        StorageAccessor storageAccessor = new InMemoryStorage();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1144]: nit: i'd move this line right before using it when declaring the federationStorageProvider

**Review by julia-zack — APPROVED** [review]
lgtm! Left two nit comments

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -1138,6 +1138,30 @@ void save_savePendingFederation_preRSKIP123_shouldBeSavedInStorageSerializedFrom
 
     }
 
+    @Test
+    void saveActiveFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNullAndRSKIP186IsActive() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1142]: nit: reverse the order of the name, first the `whenHeightIsNotNullAndRSKIP186IsActive` condition and then the `shouldSaveToStorage` expected behaviour

**Review by wilmerrootstock — APPROVED** [review]
Well done!

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1144]: I wonder how that happened 🤔🤣

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1142]: I wonder how that happened 🤔🤣

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -1138,6 +1138,30 @@ void save_savePendingFederation_preRSKIP123_shouldBeSavedInStorageSerializedFrom
...
+        // Arrange
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        long expectedFederationCreationBlockHeight = 100;
+        federationStorageProvider.setActiveFederationCreationBlockHeight(expectedFederationCreationBlockHeight);
+
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

**Pull request closed without merging**